bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -115,4 +129,70 @@
      */
     public void runNow(String ruleUID);
 
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for a given scope or {@code null} scope.
+     * @param scope to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasScope(String scope) {
+        if (null == scope) {
+            return r -> null == r.getScope();
+        } else {
+            return r -> scope.equals(r.getScope());
+        }
+    }
+
+    /**
+     * Creates a {@link Predicate} which can be used to filter {@link Rule}s for any of the given scopes and even {@code null} scope.
+     * @param scopes to search for
+     * @return created {@link Predicate}
+     */
+    public static Predicate<Rule> hasAnyScope(String... scopes) {
+        HashSet<String> scopeSet = new HashSet<String>(scopes.length);
```

Comment:
Implementation. This can be impl by calling getAll(Predicate)  and should not be part of the interface. It is related to your business logic.